WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.m

```diff
@@ -471,15 +480,81 @@ - (void)trackAppearedIfNeeded
     self.trackedViewDisplay = YES;
 }
 
-- (void)disableInteractionsForNotification:(Notification *)note
+
+#pragma mark - Undelete Mechanism
+
+- (void)showUndeleteForNote:(Notification *)note onTimeout:(NotificationDetailsDeletionActionBlock)onTimeout
+{
+    // Locate the IndexPath
+    NSIndexPath *indexPath = [self.tableViewHandler.resultsController indexPathForObject:note];
+    if (!indexPath) {
+        return;
+    }
+    
+    // Keep a copy of the Destructive Block
+    [self.notificationIdsWithPendingDeletion addObject:note.objectID];
+    
+    // Update the cell
+    [self.tableView reloadRowsAtIndexPaths:@[ indexPath ] withRowAnimation:UITableViewRowAnimationFade];
+    
+    // Dispatch the Action block
+    dispatch_time_t timeout = dispatch_time(DISPATCH_TIME_NOW, (int64_t)(NotificationsUndoTimeout * NSEC_PER_SEC));
+    dispatch_after(timeout, dispatch_get_main_queue(), ^{
+        [self performPendingDeletionAction:onTimeout noteObjectID:note.objectID];
```

Comment:
Would it be slightly safer to assign the objectID to a var and use it in the dispatch block vs using the note in the block?